website/docs/plugin/terraform-plugin-protocol.mdx

```diff
@@ -0,0 +1,227 @@
+- UpgradeResourceState
+  - only called if resource already exists in state
+- [ValidateResourceConfig](#validateconfig-rpcs)
+- [ValidateDataResourceConfig](#validateconfig-rpcs)
+- [ReadResource](#read-rpcs)
+  - only called if resource already exists in state)
+- [ReadDataSource](#read-rpcs)
+- [PlanResourceChange](#planresourcechange-rpc)
+  - if _-auto-approve_ is not used this is the point at which Terraform prompts the practitioner to confirm the proposed changes
+- [GetProviderSchema](#getproviderschema-rpc)
+- [ValidateProviderConfig](#validateconfig-rpcs)
+- [ConfigureProvider](#configureprovider-rpc)
+- UpgradeResourceState
+  - only called if resource already exists in state
+- [ValidateResourceConfig](#validateconfig-rpcs)
+- [PlanResourceChange](#planresourcechange-rpc)
+- [ApplyResourceChange](#applyresourcechange-rpc)
+
+The correlation between the RPCs and the Terraform plugin framework methods is as follows:
+
+| Protocol RPCs                                                                     | Framework Functionality                                                                                                                                            |
+|-----------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `GetProviderSchema`                                                               | `provider.Provider`, `resource.Resource`, and `datasource.DataSource` interface `Schema` method                                                                    |
+| `ValidateProviderConfig` / `ValidateResourceConfig` /`ValidateDataResourceConfig` | `provider.Provider`, `resource.Resource`, and `datasource.DataSource` interface `Schema` method validators, `ConfigValidators` method, and `ValidateConfig` method |
+| `ConfigureProvider`                                                               | `provider.Provider` interface `Configure` method                                                                                                                   |
+| `ReadResource` / `ReadDataSource`                                                 | `resource.Resource` and `datasource.DataSource` interface `Read` method                                                                                            |
+| `PlanResourceChange`                                                              | `resource.Resource` interface `Schema` method plan modifiers and `ModifyPlan` method                                                                               |
+| `ApplyResourceChange`                                                             | `resource.Resource` interface `Create`, `Update`, or `Delete` method                                                                                               |
+
+## RPCs and Terraform plugin framework
```

Comment:
This seems like it should be its own framework documentation page since its SDK specific and could change over time.